Medicina/Misc/MedScripts/src/med-vnc

```diff
@@ -0,0 +1,171 @@
+    done
+    exit 0
+    ;;
+    -p|--port)
+    ssh_port=$2
+    shift
+    shift
+    ;;
+    -v|--viewonly)
+    viewonly="-ViewOnly"
+    shift
+    ;;
+    *)
+    if [[ $1 == *"@"* ]]; then
+        while IFS='@' read -ra ARG; do
+            if [[ "${#ARG[@]}" -ne 2 ]]; then
+                echo "Please provide a correct <role>@<gatein> pair, retry!"
+                print_help
+            fi
+            role=${ARG[0]}
+            gatein=${ARG[1]}
+        done <<< "$1"
+    else
+        echo "Unrecognized option: '$1'!"
+        print_help
+    fi
+    shift
+    ;;
+esac
+done
+
+if [[ "$gatein" == "local" ]]; then
+    local_connection="TRUE"
+fi
+
+if [[ "$OSTYPE" == *"linux-gnu"* ]]
+then
+   echo -e "Linux OS is detected..."
+   current_system="LINUX"
+elif [[ "$OSTYPE" == *"darwin"* ]]
+then
+   echo -e "Mac OS is detected..."
+   current_system="MACOS"
+else
+   echo -e "Not supported OS, some unpredictables results may happen!"	
+fi
+
+if ! hash vncviewer &>/dev/null; then
+    echo -e "'vncviewer' is not installed, this script is setting up the connection but won't automatically start vnc.\nA custom client should be manually started."
+    vnc_avail="FALSE"
+fi
+
+if hash netstat &>/dev/null; then
+    if [[ "$current_system" == "LINUX" ]]; then
+        PORTCOMMAND="netstat -tulpn"
+    else
+        PORTCOMMAND="netstat -avnp tcp"
+    fi
+else
+    echo -e "This program requires 'netstat' to be installed.\nThe package to install may vary depending on your operating system.\n"
+    print_help
+fi
+
+for session in "${remote_sessions[@]}" ; do
+    role_name="${session%%:*}"
+    ports="${session#*:}"
+    rport="${ports%:*}"
+    lport="${ports#*:}"
+    if [[ "$role" == "$role_name" ]]; then
+        if [ "$local_connection" = "TRUE" ]; then
+        		#this is for local area connection when it will enabled
+            #remote_port=$lport
+            remote_port=$rport
+            ssh_port=5122 
+        else
+            remote_port=$rport
+        fi
+        break
+    fi
+done
+
+if [[ $remote_port -eq 0 ]]; then
+    echo "Please, provide a correct role, use the switch '--roles' for a complete list of available roles"
+    print_help
+fi
+
+for i in {0..50}
+do
+    local_port_attempt=$((9000 + i))
+    if ! $PORTCOMMAND | grep 127.0.0.1[.:]$local_port_attempt &>/dev/null; then
+        local_port=$local_port_attempt
+        break
+    fi
+done
+
+if [[ -z "$local_port" ]]; then
+    echo "Could not find an available port in range 9000-9050"
+    exit 1
+fi
+
+#echo $local_connection
+#echo $role
+#echo $remote_port
+#echo $local_port
+#echo $gatein
+#echo $ssh_port
+#echo $viewonly
+#echo $vnc_avail
+
+if [[ "$local_connection" = "TRUE" ]]; then
+    echo "Connecting from local area...."
+    #ssh -N -f -L $local_port:192.168.1.99:$remote_port $role@192.167.189.98
+else
+    echo "Connecting from wide area....."
+fi
+ssh -N -f -p $ssh_port -L $local_port:192.167.189.98:$remote_port $role@$gatein
```

Comment:
Shouldn't this be inside the previous `else` block? If I remember correctly, the vnc session needs a ssh tunnel only when connecting from outside (in LAN the vnc should be directly accessible, at least with the latest updates on the vnc servers). I also understand that this script will be used when connecting from the inside by connecting to the gateway and not directly to the desired machine. In that case this line is correct to be here.

Reply:
That's correct the idea is to directly connecting to discos when insider the local network, but this needs a configuration in the proxy machine that cannot be done right now (until the old escs system is decommissioned), until then we go through the gatein also from LAN. I also received some more comments on theses script si i'm going to have some changes to cope with them. I'll send un update soon.